### PR TITLE
Add 'fill' visual

### PR DESCRIPTION
**Pull request by daliasen**

### CODE REVIEW

File: server/controller.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import time
 import generators
 from interactive import draw, snake, write, interactive_autoscroll, spectrogram, rhythm
-from visuals import corners, cube_visuals, edges, extend, faces, fireworks, flash, flatten, layers, line, matrix, pins, pulse, rainbow, rope, rotate, shuffle, snakes, spiral, starfield, tetris, text, wave
+from visuals import corners, cube_visuals, edges, extend, faces, fireworks, fill, flash, flatten, layers, line, matrix, pins, pulse, rainbow, rope, rotate, shuffle, snakes, spiral, starfield, tetris, text, wave
```
Comment (abryant) [inline server/controller.py]: nit: alphabetical order

File: server/controller.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
   'faces': faces.faces,
   'fireworks': fireworks.fireworks,
   'flash': flash.flash,
+  'fill': fill.fill,
```
Comment (abryant) [inline server/controller.py]: nit: before fireworks here too

File: server/static/index.html [hunk 3]
```diff
@@ -54,6 +54,7 @@ <h1>Cube Controller</h1>
           <button type='button' class='btn' onclick='start("extend")'>Extend</button>
           <button type='button' class='btn' onclick='start("faces")'>Faces</button>
           <button type='button' class='btn' onclick='start("fireworks")'>Fireworks</button>
+          <button type='button' class='btn' onclick='start("fill")'>Fill</button>
```
Comment (abryant) [inline server/static/index.html:56]: nit: and here

File: visuals/fill.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from .cube import *
+import random
+
+def fill():
+  c = Cube()
+  colour = Colour.cyan()
+  on = True
+  while True:
+    all_coords = [(x, y, z) for x in range(c.size) for y in range(c.size) for z in range(c.size)]
```
Comment (abryant) [inline visuals/fill.py]: nit: you could use Pos(x, y, z) here instead of below

File: visuals/fill.py [hunk 5]
```diff
@@ -3,16 +3,17 @@
...
     all_coords = [(x, y, z) for x in range(c.size) for y in range(c.size) for z in range(c.size)]
     random.shuffle(all_coords)
-    for coord in all_coords:
+    for counter, coord in enumerate(all_coords):
         if on:
           c.set(Pos(*coord), colour)
         else:
           c.set(Pos(*coord), Colour.BLACK)
-        yield c.copy()
+        if counter % c.size == 7:
```
Comment (abryant) [inline visuals/fill.py]: nit: c.size - 1

**Review by abryant — COMMENTED** [review]
Thanks! :D

File: server/controller.py (hunk 1 above)
Comment (daliasen) [reply server/controller.py]: done!

File: server/controller.py (hunk 2 above)
Comment (daliasen) [reply server/controller.py]: done!

File: server/static/index.html (hunk 3 above)
Comment (daliasen) [reply server/static/index.html:56]: done!

File: visuals/fill.py (hunk 4 above)
Comment (daliasen) [reply visuals/fill.py]: done!

File: visuals/fill.py (hunk 5 above)
Comment (daliasen) [reply visuals/fill.py]: done! thanks :)

**Pull request merged**